packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,216 @@
+/*************************************************************************
+ * ADOBE CONFIDENTIAL
+ * ___________________
+ *
+ * Copyright 2024 Adobe
+ * All Rights Reserved.
+ *
+ * NOTICE: All information contained herein is, and remains
+ * the property of Adobe and its suppliers, if any. The intellectual
+ * and technical concepts contained herein are proprietary to Adobe
+ * and its suppliers and are protected by all applicable intellectual
+ * property laws, including trade secret and copyright laws.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Adobe.
+ **************************************************************************/
+
+import {CheckboxGroup, Label, Provider, RadioGroup} from 'react-aria-components';
+import {IconContext, TextContext} from '@react-spectrum/s2';
+import React, {
+  ForwardedRef,
+  forwardRef,
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useState
+} from 'react';
+import {style}  from '../style' with {type: 'macro'};
+
+/**
+ * Ensures the return value is a string.
+ * @param { string | string[] | undefined } value Possible options for selection.
+ * @returns { string }
+ */
+function unwrapValue(value: string | string[] | undefined): string {
+  if (Array.isArray(value)) {
+    return value[0];
+  }
+  return value;
+}
+
+/**
+ * Ensures the return value is an array or undefined.
+ * @param {string | string[] | undefined} value Possible options for selection.
+ * @returns { [] | undefined }
+ */
+function ensureArray(value: string | string[]): string[] {
+  if (Array.isArray(value)) {
+    return value;
+  }
+  return value ? [value] : [];
+}
+
+export {unwrapValue, ensureArray};
+
+export type SelectBoxValueType = string | string[];
```

Comment:
i think we can simplify this to always an array, that's how our other components that can switch single/multi select work

Reply:
@snowystinger do you have guidance/recommendations for whether this component should be using `ValueBase` for props `value` and `onChange` or a selection for props `selectedKeys` and `selectionMode`?